src/packet/public_key.js

```diff
@@ -100,8 +102,11 @@ PublicKey.prototype.read = function (bytes) {
     var p = 0;
 
     for (var i = 0; i < mpicount && p < bmpi.length; i++) {
-
-      this.mpi[i] = new type_mpi();
+      if (this.algorithm === 'ecdsa' && i === 0) {
```

Comment:
Here is the issue related to the assumption that key packets only contain MPI. Also for 'ecdh' I've to add a similar condition.